### ISSUE TITLE
cmdstan unhappy with generated stan code

### ISSUE THREAD

**Issue by karldw** [paul-buerkner/brms#926]
Thanks for the wonderful package! I was trying out the new cmdstan backend, and I noticed that models would sometimes fail to compile. Maybe this is because I'm using a newer version of Stan?

Here's an example:
```r

library(brms)
library(cmdstanr)

temp_stan <- tempfile(fileext=".stan")

# The make call fails silently within brms, but loudly if we compile via cmdstanr::model
fit1 <- brm(count ~ zAge + zBase * Trt + (1|patient), 
  data = epilepsy, family = poisson(), 
  backend="cmdstan", save_model = temp_stan)
exists("fit1")
# [1] FALSE
  
mod <- cmdstanr::cmdstan_model(temp_stan)
# Compiling Stan program...
# Error in processx::run(command = make_cmd(), args = c(tmp_exe, cpp_options_to_compile_flags(cpp_options),  : 
#   System command 'make' failed, exit status: 2, stderr (last 10 lines):
# E>     46:      - 1 * student_t_lccdf(0 | 3, 0, 2.5);
# E>     47:    target += std_normal_lpdf(z_1[1]);
# E>                      ^
# E>     48:    // likelihood including all constants
# E>     49:    if (!prior_only) {
# E>    -------------------------------------------------
# E> 
# E> Probabilty functions with suffixes _lpdf, _lpmf, _lcdf, and _lccdf, require a vertical bar (|) between the first two arguments.
# E> 
# E> make: *** No rule to make target '/tmp/Rtmp1VLICk/model-4acf14844fa2.hpp', needed by '/tmp/Rtmp1VLICk/model-4acf14844fa2'.  Stop.

```

Versions: 
- brms 2.12.3 (54bca34)
- cmdstan 2.23.0
- cmdstanr 0.0.0.9002

**Comment by paul-buerkner** [comment]
This is a cmdstan bug that should have been fixed already on github. @jgabry can you confirm?

**Comment by rok-cesnovar** [comment]
Yes, right now, that needs to be

`target += std_normal_lpdf(z_1[1]|);`

The | after the first arg is mandatory (for now).

**Comment by jgabry** [comment]
@rok-cesnovar I guess an alternative would be to compile with `cpp_options = list("STANC2"=TRUE)`, right?

**Comment by rok-cesnovar** [comment]
Yes, that would also work. This should be fixed for 2.24.

**Comment by paul-buerkner** [comment]
Thanks @jgabry and @rok-cesnovar for you quick help. Since it is not a brms bug, I will close this issue.
